### PR TITLE
Integrating Azure authentication

### LINKED ISSUE #4: Integrate Azure KeyVault + secrets management
Properly incorporate secrets and KeyVault access so that the app can pull schemas from blob storage instead of locally. The app should be able to access our Azure blob storage from within Docker. Ensure no secrets are exposed in the repo or code.

### PR DESCRIPTION
**Pull request by amondal2**
Closes #4 --

First pass at integrating Azure secrets into the application. Since the application will be interacting with Azure via Docker, we only use Service Principal credentials to authenticate (as opposed to individual user credentials). 

This PR:
1. Adds authentication utils to pull environment variables and authenticate
2. Injects secrets into the Docker container via docker-compose and an .env file (not committed)
3. Adds an auth route to test if the application is authenticated properly 

I looked into Docker secrets (https://docs.docker.com/engine/swarm/secrets/) for this, but it doesn't seem to me is much more secure given the secrets are just written to disk instead of as env variables (and more annoying to read from python), and if someone is able to ssh into the containers on the CDC network, we probably have bigger issues. Happy to take suggestions on how to improve the workflow, in any case.

I'll open separate issues/PRs for actually pulling from blob storage and automated deployments.

### CODE REVIEW

File: validation-api/azure_utils/auth.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Since this will be run from a Docker container,
+    # we only use the SP Credential to Authenticate.
+
+    # Pull values from injected environment variables
+    tenant_id = os.environ.get("TENANT_ID", "")
+    application_id = os.environ.get("APPLICATION_ID", "")
+    client_secret = os.environ.get("SP_SECRET", "")
+
+    sp_credential = ClientSecretCredential(
```
Comment (natemcintosh) [inline validation-api/azure_utils/auth.py:37]: what happens when an empty string gets passed into `ClientSecretCredential()`? I'm guessing it just errors out? But maybe would be nice to have a custom error message explaining that one of the arguments was an empty string

**Review by natemcintosh — APPROVED** [review]

**Review by zsusswein — APPROVED** [review]
To make sure I'm following: the requesting user doesn't provide anything to authenticate, right?

We make sure the server is provisioned with the appropriate env var credentials. So a call to `auth()` is really a check that the server has been appropriately set up? (or "server" in this case?)

**Comment by amondal2** [comment]
@zsusswein yep, that's correct!

File: validation-api/azure_utils/auth.py (hunk 1 above)
Comment (amondal2) [reply validation-api/azure_utils/auth.py:37]: good call @natemcintosh -- I'll integrate this check!

**Review by natemcintosh — APPROVED** [review]

**Pull request merged**